datadog/resource_datadog_security_monitoring_filters.go

```diff
@@ -0,0 +1,253 @@
+			},
+			"is_enabled": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether the security filter is enabled.",
+			},
+			"exclusion_filter": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Exclusion filters to exclude some logs from the security filter.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Exclusion filter name.",
+						},
+						"query": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Exclusion filter query. Logs that match this query are excluded from the security filter.",
+						},
+					},
+				},
+			},
+
+			"filtered_data_type": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Description:      "The filtered data type.",
+				Default:          "logs",
+				ValidateDiagFunc: validators.ValidateEnumValue(datadogV2.NewSecurityFilterFilteredDataTypeFromValue),
+			},
+		},
+	}
+}
+
+func resourceDatadogSecurityMonitoringFilterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV2 := providerConf.DatadogClientV2
+	authV2 := providerConf.AuthV2
+
+	filterCreate := buildSecMonFilterCreatePayload(d)
+
+	response, httpResponse, err := datadogClientV2.SecurityMonitoringApi.CreateSecurityFilter(authV2, *filterCreate)
+	if err != nil {
+		return utils.TranslateClientErrorDiag(err, httpResponse, "error creating security monitoring filter")
+	}
+
+	// store the resource id
+	data := response.GetData()
+	d.SetId(data.GetId())
+
+	return nil
+}
+
+func resourceDatadogSecurityMonitoringFilterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV2 := providerConf.DatadogClientV2
+	authV2 := providerConf.AuthV2
+
+	id := d.Id()
+	filterResponse, httpResponse, err := datadogClientV2.SecurityMonitoringApi.GetSecurityFilter(authV2, id)
+	if err != nil {
+		if httpResponse != nil && httpResponse.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	updateResourceDataFilterFromResponse(d, filterResponse)
+
+	// handle warning
+	if filterResponse.HasMeta() {
+		filterMeta := filterResponse.GetMeta()
+		warning := filterMeta.GetWarning()
+		diagnostic := diag.Diagnostic{Severity: diag.Warning, Summary: warning}
+		return diag.Diagnostics{diagnostic}
+	}
+	return nil
+}
+
+func resourceDatadogSecurityMonitoringFilterUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV2 := providerConf.DatadogClientV2
+	authV2 := providerConf.AuthV2
+
+	filterId := d.Id()
+
+	response, diagnostics, failed := fetchFilterFromApi(datadogClientV2, authV2, filterId)
```

Comment:
the purpose of that call is to get the latest version of the filter; we mostly need to fetch the value of "version" field; to set to to the latest one; otherwise we can't issue a proper update.

Reply:
It seems to me that there should be a `version` computed attribute then, that you update in read. Then you could pass it directly again.

Reply:
Will read be called before an update ?

Reply:
We did the same for the rule ressource actually.

Reply:
It's done for the default rule, not the normal one for some reason. I don't think it's correct either. But yes read is called during planning before any operation.